client/blocks/edit-gravatar/index.jsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+import debugFactory from 'debug';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FilePicker from 'components/file-picker';
+import FormLabel from 'components/forms/form-label';
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getToken as getOauthToken } from 'lib/oauth-token';
+import Gravatar from 'components/gravatar';
+import {
+	isCurrentUserUploadingGravatar,
+} from 'state/current-user/gravatar-status/selectors';
+import { isOffline } from 'state/application/selectors';
+import Spinner from 'components/spinner';
+import { uploadGravatar } from 'state/current-user/gravatar-status/actions';
+
+/**
+ * Module dependencies
+ */
+const debug = debugFactory( 'calypso:edit-gravatar' );
+
+export class EditGravatar extends Component {
+	constructor() {
```

Comment:
Good point. I'll be adding state in an upcoming PR, but I can add it in all in one go then. Removed for now.